### ISSUE TITLE
All Traffic widget pie chart legend should not be cut off

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#2700]
The legend entries on the new All Traffic widget pie chart currently get shortened if there is not enough space available, and sometimes they even receive an arrow to see the other ones. See screenshot below:
![Screen Shot 2021-01-27 at 11 07 42 AM](https://user-images.githubusercontent.com/3531426/106040979-10e60800-6090-11eb-9899-fba03010d18e.png)

Instead of this, they should always show completely and never be shortened/cut off. If they take too much horizontal space, they should instead float into a new line below. We intentionally limit the results to 5, so this should never result in a crazy amount of vertical space.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The legend entries on the new All Traffic widget pie chart should always be shown completely and never be shortened with an ellipsis (unless a single one would be longer than the available horizontal space for the entire pie chart, which is highly unlikely). They should float into multiple rows as needed. In accordance with that, there should also never be an arrow to see the other entries that didn't fit.

#### DashboardAllTrafficWidget/index.js:
* Pass the `dimensionValue` variable to the `UserDimensionsPieChart` element as we do it for the `TotalUserCount` element;

#### DashboardAllTrafficWidget/UserDimensionsPieChart.js:
* Add a new property `dimensionValue` to the `UserDimensionsPieChart.propTypes` map;
* Disable the built-in legend in the user dimensions pie chart by setting the `position: 'none'` setting for the `legend` options;
* Create a new `onLegendClick` callback in the `UserDimensionsPieChart` component:
  * The callback should accept one argument: an index of clicked legend item;
  * Grab the chart instance using the `GoogleChart.charts.get( chartID )` function (see how we do it in the onSelect handler);
  * Get the current selection using `chart.getSelection()` function;
  * Call `chart.setSelection( null )` if there is already a selected row with an index that equals to the clicked legend item index, otherwise call `chart.setSelection( [ { row: selectedItem, column: null } ] )`;
  * Dispatch the new value for the `dimensionValue` property as we do it in the onSelect handler.
* Render the new legend beneath the google chart element using the new `onLegendClick` callback for legend items. It can be done something like this:
  ```javascript
  <div>
      { dataMap.slice( 1 ).map( ( [ label ], i ) => (
          <button key={ label } onClick={ () => onLegendClick( i ) }>{ label }</button>
      ) ) }
  </div>
  ```
* Add styles to make the new legend look like in the original [Figma file](https://www.figma.com/file/NzvJTPFdqUVJC9Q1mmih4e/Google-SK-UX-Enhancements10up?node-id=1198%3A271).
  * Don't forget to add a background color to the selected legend item.

### Test Coverage

* N/A

### Visual Regression Changes

* The legend of the user dimensions pie chart should look like on the original Figma file.

## QA Brief

* Set up the site kit on your local;
* Activate and connect the Analytics module;
* Go to the dashboard page;
* Make sure you see the legend beneath the pie chart in the all traffic widget;
* Make sure the legend items aren't cut off on smaller screens;
* Make sure that when you click on a legend item, the appropriate pie slice is selected and the user count graph is updated accordingly;
* Make sure when you click on the same legend item, the selected pie slice is deselected and the user count graph gets back to its original state;
* Make sure the design of the new legend matches the original Figma file including the selected state for the legend item.

## Changelog entry

* Enhance All Traffic widget pie chart legend by using a custom implementation so that the labels are never cut off and include a visual hover and active state.

**Comment by felixarntz** [comment]
IB ✅ 

Great that we'll get rid of the regular legend here, which will also allow us to use the original highlighting from the Figma file.

**Comment by wpdarren** [comment]
### QA Update: Fail ❌ 

@asvinb @aaemnnosttv there are a few issues. I was able to recreate these on all the 3 dimensions. 

1) When I click on the first legend item, i.e `organic search`.  The graph displays the correct number of users but the line chart is blue rather than yellow in this example. It only seems to be an issue on the first item because the others change to the correct color when you select and unselect. 

![image](https://user-images.githubusercontent.com/73545194/107519939-d2059580-6beb-11eb-809d-35b54c562a6e.png)

2) If you select a legend item, let's say `direct`, but do not deselect, instead select another item like `email` the results of the line chart display the email data, but the color of the line charge is `direct` in this case purple, instead of blue. 

![image](https://user-images.githubusercontent.com/73545194/107521025-0463c280-6bed-11eb-8e24-af9073ccfea9.png)

**Comment by asvinb** [comment]
@aaemnnosttv Create a new PR to fix the issue that @wpdarren noted.

**Comment by aaemnnosttv** [comment]
@wpdarren – this should be fixed now, can you check again once this job is finished?
https://github.com/google/site-kit-wp/actions/runs/555006608

**Comment by wpdarren** [comment]
### QA Update: Fail ❌ 

@aaemnnosttv @asvinb I have a few observations which I'd like you to look at. 

`Make sure that when you click on a legend item, the appropriate pie slice is selected and the user count graph is updated accordingly` 

When a legend item is selected, on smaller screen sizes (i.e iPhone, Android) the pie chart slice highlight is getting cut off. The issue occurs if you click on the slice or the legend item so it might not be linked with this issue.  [Screenshot 1](https://user-images.githubusercontent.com/73545194/107594995-69033980-6c4e-11eb-80d1-2102b4d6d598.png)  / [Screenshot 2](https://user-images.githubusercontent.com/73545194/107595019-7a4c4600-6c4e-11eb-8140-74d6776441ea.png)

`Make sure when you click on the same legend item, the selected pie slice is deselected and the user count graph gets back to its original state`

When you deselect the legend item, the selected pie slice is deselected but the hover color of the legend item still appears.  Should this be the case? i.e. I deselect 'Email' the Email slice is deselected, but you can still see a color on the item legend, even though it appears a lighter shade. [Screenshot](https://user-images.githubusercontent.com/73545194/107595413-7c62d480-6c4f-11eb-9f07-6960776973a3.png)

**Verified:**

- The legend appears beneath the pie chart in the all traffic widget. - [Screenshot](https://user-images.githubusercontent.com/73545194/107594753-aca97380-6c4d-11eb-918a-4de0fd0e2650.png)
- The legend items aren't cut off on smaller screens  - [Screenshot](https://user-images.githubusercontent.com/73545194/107594753-aca97380-6c4d-11eb-918a-4de0fd0e2650.png)
- The design of the new legend matches the original Figma file including the selected state for the legend item.

**Comment by asvinb** [comment]
@wpdarren @aaemnnosttv 

- For the first issue, it's not related to the PR. I suggest we create another ticket to tackle it.
- The second issue is a non issue since you are seeing the hover state (where the font weight is normal, compared to when it's selected, the font weight is 500). This is completely normal, since when deselecting the legend, your mouse is actually on the item.

**Comment by wpdarren** [comment]
@asvinb thank you, I will submit a separate ticket for issue 1) and issue 2) I was just about to update my comment to say exactly what you've noted. I'll get this moved forward. Thank you.

**Comment by wpdarren** [comment]
### QA Update: Pass ✅ 

**Verified:**

- The legend appears beneath the pie chart in the all traffic widget. - [Screenshot](https://user-images.githubusercontent.com/73545194/107594753-aca97380-6c4d-11eb-918a-4de0fd0e2650.png)
- The legend items aren't cut off on smaller screens  - [Screenshot](https://user-images.githubusercontent.com/73545194/107594753-aca97380-6c4d-11eb-918a-4de0fd0e2650.png)
- When you click on the same legend item, the selected pie slice is deselected and the user count graph gets back to its original state [Screenshot](https://user-images.githubusercontent.com/73545194/107612643-a503c200-6c81-11eb-8008-bb8d1f0c919e.png)

- The design of the new legend matches the original Figma file including the selected state for the legend item
